CRM/ManualDirectDebit/Hook/Alter/ContactDetailReport.php

```diff
@@ -0,0 +1,157 @@
+   *
+   * @return bool
+   */
+  protected function shouldUpdate($reportForm) {
+    if (
+      isset($reportForm->getVar('_params')['fields']['dd_instruction_batch_id'])
+      || isset($reportForm->getVar('_params')['dd_instruction_batch_id_value'])
+      || isset($reportForm->getVar('_params')['dd_instruction_batch_id_op'])
+    ) {
+      return TRUE;
+    }
+    return FALSE;
+  }
+
+  /**
+   * Update column list
+   *
+   * @param array $columns
+   */
+  protected function updateColumns(&$columns) {
+    $batches = $this->getBatches();
+    $columns['civicrm_value_dd_mandate']['fields']['dd_instruction_batch_id'] = [
+      'title' => ts('Instruction Batch'),
+      'dbAlias' => "instruction_batches.batch_id",
+    ];
+
+    $columns['civicrm_value_dd_mandate']['filters']['dd_instruction_batch_id'] = [
+      'title' => ts('Instruction Batch'),
+      'dbAlias' => 'instruction_batches.batch_id',
+      'type' => CRM_Utils_Type::T_INT,
+      'operatorType' => CRM_Report_Form::OP_MULTISELECT,
+      'options' => $batches,
+    ];
+  }
+
+  /**
+   * update the SQL Query
+   *
+   * @param CRM_Report_Form_Contact_Detail $reportForm
+   */
+  protected function updateSql(&$reportForm) {
+    if (!$this->shouldUpdate($reportForm)) {
+      return;
+    }
+
+    $from = $reportForm->getVar('_from');
+
+    // prevent double left join with civicrm_value_dd_mandate
+    if (strpos($from, 'civicrm_value_dd_mandate') === FALSE) {
+      $from .= "
+          LEFT JOIN civicrm_value_dd_mandate value_dd_mandate_civireport
+          ON (value_dd_mandate_civireport.entity_id = contact_civireport.id)
+        ";
+    }
+
+    $from .= "
+        LEFT JOIN civicrm_entity_batch instruction_batches
+        ON (instruction_batches.entity_table = 'civicrm_value_dd_mandate'
+        AND instruction_batches.entity_id = value_dd_mandate_civireport.id)
+      ";
+
+    $reportForm->setVar('_from', $from);
+  }
+
+  /**
+   * Update rows for display
+   *
+   * @param array $rows
+   */
+  protected function updateRows(&$rows) {
+    $batches = $this->getBatches();
+    foreach ($rows as $rowNum => $row) {
+      if (isset($rows[$rowNum]['civicrm_value_dd_mandate_dd_instruction_batch_id'])) {
+        $rows[$rowNum]['civicrm_value_dd_mandate_dd_instruction_batch_id'] = $batches[$row['civicrm_value_dd_mandate_dd_instruction_batch_id']] ?? NULL;
+      }
+    }
+  }
+
+  /**
+   * Get Batches
+   *
+   * @return array $batches
+   */
+  protected function getBatches() {
+    if (count($this->_batches)) {
+      return $this->_batches;
+    }
+
+    $condition = " AND (
+      v.name = '" . BatchHandler::BATCH_TYPE_INSTRUCTIONS . "'
+      OR v.name = '" . BatchHandler::BATCH_TYPE_PAYMENTS . "'
+      OR v.name = '" . BatchHandler::BATCH_TYPE_CANCELLATIONS . "'
+    )";
+    $batchTypeIds = CRM_Core_OptionGroup::values('batch_type', FALSE, FALSE, FALSE, $condition, 'value');
+
+    $params = [
+      'type_id' => ['IN' => $batchTypeIds],
```

Comment:
civicrm API will retreuve only 25 records by default, so you need to pass this : 

```
$params = [
'type_id' => ['IN' => $batchTypeIds],
'options' => ['limit' => 0],
..etc
]
```

to remove this restriction